climada/hazard/centroids/centr.py

```diff
@@ -491,52 +491,83 @@ def clear(self):
+            self.set_meta_to_lat_lon()
+        if centr.meta and not centr.lat.any():
+            centr.set_meta_to_lat_lon()
+
+        if not u_coord.equal_crs(centr.geometry.crs, self.geometry.crs):
+            raise ValueError('Different CRS not accepted.')
+
+        self.meta = dict()
+        self.__dict__.update(self.union(centr).__dict__)
+
+
+    def union(self, *others):
+        """
+        Create the union of centroids from the inputs.
+
+        The centroids are combined together point by point.
+        Rasters are converted to points and raster information is lost.
+        All centroids must have the same CRS.
+
+        Parameters
+        ----------
+        others : any number of climada.hazard.Centroids()
+            Centroids to form the union with
+
+        Returns
+        -------
+        centroids : climada.hazard.Centroids()
+            Centroids containing the union of the centroids in centroids_list.
+
+        """
+
+        if self.lat.any() or self.meta:
+            centroids = copy.deepcopy(self)
         else:
-            LOGGER.debug('Appending points')
-            if not u_coord.equal_crs(centr.geometry.crs, self.geometry.crs):
-                raise ValueError('Different CRS not accepted.')
-            self.lat = np.append(self.lat, centr.lat)
-            self.lon = np.append(self.lon, centr.lon)
-            self.meta = dict()
-
-        # append all 1-dim variables
-        for (var_name, var_val), centr_val in zip(self.__dict__.items(),
-                                                  centr.__dict__.values()):
-            if isinstance(var_val, np.ndarray) and var_val.ndim == 1 and \
-            var_name not in ('lat', 'lon'):
-                setattr(self, var_name, np.append(var_val, centr_val).
-                        astype(var_val.dtype, copy=False))
+            centroids = copy.deepcopy(others[0])
+            others = others[1:]
+        if centroids.meta and not centroids.lat.any():
+            centroids.set_meta_to_lat_lon()
+
+        centroids.set_geometry_points()
+        for cent in others:
+            if cent.meta and not cent.lat.any():
+                cent.set_meta_to_lat_lon()
+            cent.set_geometry_points()
+            if not u_coord.equal_crs(cent.geometry.crs,
+                                      centroids.geometry.crs):
+                    raise ValueError('Different CRS are not accepted.')
+            if not centroids.equal(cent):
+                # append all 1-dim variables
+                for (var_name, var_val), centr_val in \
+                    zip(centroids.__dict__.items(), centroids.__dict__.values()):
+                    if isinstance(var_val, np.ndarray) and var_val.ndim == 1:
+                        setattr(centroids, var_name,
+                                np.append(var_val, centr_val).astype(
+                                    var_val.dtype, copy=False
+                                    )
```

Comment:
Don't use `np.append` iteratively, but collect all the data and apply `np.cocatenate` at the end.

Reply:
Ok, I'll see to do that. I copied that bit blindly from the older code.